returnn/util/basic.py

```diff
@@ -3180,6 +3180,7 @@ def __init__(self, base_name, code_version, code,
     :param bool should_cleanup_old_all: whether we should look in the cache dir
       and check all ops if we can delete some old ones which are older than some limit (self._cleanup_time_limit_days)
     :param bool should_cleanup_old_mydir: whether we should delete our op dir before we compile there.
+    :param io.TextIOBase log_stream: file stream for print statements, defaults to sys.stdout
```

Comment:
Type is wrong. At least `...|None`. Or maybe `io.TextIOBase|io.StringIO|typing.TextIO|None`.

Reply:
`io.StringIO` and `typing.TextIO` are not needed, as they are inheriting `io.TextIOBase`.

Reply:
I think you should better use `typing.TextIO`, as this is more generic (the inheritance is not correct / not relevant I think).
I think `io.TextIOBase` is missing some functions as well, so you might get warnings elsewhere.